stdlib/public/SDK/Foundation/Data.swift

```diff
@@ -613,9 +613,9 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
         }
     }
     
-    public subscript(bounds: Range<Index>) -> MutableRandomAccessSlice<Data> {
+    public subscript(bounds: Range<Index>) -> MutableRangeReplaceableRandomAccessSlice<Data> {
         get {
-            return MutableRandomAccessSlice(base: self, bounds: bounds)
+            return MutableRangeReplaceableRandomAccessSlice(base: self, bounds: bounds)
```

Comment:
The newly-added protocol constraint ensures that the slice type conforms to `RangeReplaceableCollection`.  I'll try to add a direct test, but I don't have access to a macOS machine.